http4s/server/src/main/scala/endpoints/http4s/server/Urls.scala

```diff
@@ -0,0 +1,196 @@
+package endpoints.http4s.server
+
+import java.net.{URLDecoder, URLEncoder}
+import java.nio.charset.StandardCharsets.UTF_8
+
+import endpoints.algebra.Documentation
+import endpoints.{PartialInvariantFunctor, Tupler, algebra}
+import org.http4s
+import org.http4s.Uri
+
+import scala.collection.compat._
+import scala.collection.mutable
+
+
+trait Urls[F[_]] extends algebra.Urls {
```

Comment:
Do we want to use an abstract type member instead of a type parameter? This would remove the burden of threading it everywhere.

~~~ scala
type Effect[A]
~~~

Also, it seems that `F` is not used here, we should probably introduce it only in `Endpoints`.

Reply:
That makes sense to me, I'll remove type parameter

Reply:
I need effect type in `Urls` since I'm returning error response when I'm decoding uri and path segments and error response needs effect type.